### PR TITLE
Authorisation tests

### PR DESCRIPTION
**Pull request by borderForNoone**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "webdriveriotypescriptrentzila",
+  "version": "1.0.0",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (AndreyCar) [inline package.json]: Remove this script

File: package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "wdio": "wdio run ./wdio.conf.ts"
```
Comment (AndreyCar) [inline package.json]: Rename to
`"tests": "wdio run ./wdio.conf.ts"`

File: package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  "keywords": ["webdriverio", "typescript"],
+  "author": "DenysKulikov",
+  "license": "ISC",
+  "description": "webdriverio typescript project",
+  "devDependencies": {
+    "@wdio/cli": "^9.0.9",
+    "@wdio/local-runner": "^9.0.9",
+    "@wdio/mocha-framework": "^9.0.8"
+  },
+  "dependencies": {
```
Comment (AndreyCar) [inline package.json]: Move dependencies from the dependencies to devDependencies

File: test/helpers/apiHelper.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import axios from 'axios';
+
+let adminAccessToken: string | null = null;
+
+export async function createAdminAccessToken() {
```
Comment (AndreyCar) [inline test/helpers/apiHelper.ts]: Create a class for API. Don't use export functions

File: test/helpers/apiHelper.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    const url = 'https://dev.rentzila.com.ua/api/auth/jwt/create/';
```
Comment (AndreyCar) [inline test/helpers/apiHelper.ts]: Create a baseURL constant and then attach an endpoint to it
``` typescript
export default class RentzilaAPI {
  private baseURL = 'https://dev.rentzila.com.ua/api';
  private adminAccessToken: string | null = null;

  createAdminAccessToken(){
    const credentials = {
        email: process.env.ADMIN_USERNAME,
        password: process.env.ADMIN_PASSWORD
    };

    try {
        const response = await axios.post(`${baseURL}/jwt/create/`, credentials);

        this.adminAccessToken = response.data.access;
        return this.adminAccessToken;
    } catch (error) {
        return error;
    }
  }
}
```

File: test/helpers/contactUsHelper.ts
Comment (AndreyCar) [inline test/helpers/contactUsHelper.ts:1]: I think we can move these functions to the HomePage class

File: test/helpers/contactUsHelper.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import homePage from '../pageobjects/home.page';
+
+export async function verifyContactUsForm() {
+    const consultationSectionNameFieldHasErrorClass = await $(homePage.consultationSectionNameField).getAttribute('class');
+    const consultationSectionPhoneNumberFieldHasErrorClass = await $(homePage.consultationSectionPhoneNumberField).getAttribute('class');
+
+    expect(consultationSectionNameFieldHasErrorClass).not.toContain('ConsultationForm_error__F1NM0');
```
Comment (AndreyCar) [inline test/helpers/contactUsHelper.ts]: Please, don't use these expects (toBe, toContain, etc). They don't wait for condition to be true. They immediately compare values. Replace this with 
 ``` typescript
await expect($(homePage.consultationSectionNameField)).not.toHaveAttr('class', /ConsultationForm_error__F1NM0/);
```

File: test/helpers/contactUsHelper.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    expect(consultationSectionPhoneNumberFieldHasErrorClass).toContain('ConsultationForm_error__F1NM0');
```
Comment (AndreyCar) [inline test/helpers/contactUsHelper.ts]: Here is the same

File: test/helpers/contactUsHelper.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+    expect($$(homePage.consultationErrorMessages)[1]).toHaveText('Телефон не пройшов валідацію');
```
Comment (AndreyCar) [inline test/helpers/contactUsHelper.ts]: Move the text to a file

File: test/pageobjects/adverts.page.ts [hunk 9]
```diff
@@ -0,0 +1,104 @@
+import Page from './page';
+
+class AdvertsPage extends Page {
+    readonly checkedCheckboxElement = '//input[@data-testid="serviceCheckbox" and @checked]/parent::div';
+    readonly relevantUnitNames = '[data-testid="serviceLabel"]';
```
Comment (AndreyCar) [inline test/pageobjects/adverts.page.ts]: Do not forget to follow the pattern of naming the selectors/elements: 
`${element type}${element name}` - > `btnSubmit`/`inputEmailAddress`/`labelDescription`/`textTitle`

File: test/pageobjects/home.page.ts [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+    readonly specialEquipmentSection = '[data-testid="specialEquipment"]';
+    readonly specialEquipmentPopularTab = '[data-testid="specialEquipment__populyarna"]';
+    readonly specialEquipmentAgriculturalTab = '[data-testid="specialEquipment__silskogospodarska"]';
+    readonly specialEquipmentBuildingTab = '[data-testid="specialEquipment__budivelna"]';
+    readonly specialEquipmentTabOther = '[data-testid="specialEquipment__insha"]';
+    readonly specialEquipmentItems = '[data-testid="specialEquipment"] [class*=RentzilaProposes_proposes_list] > *';
+    readonly specialEquipmentItemsNames = '[data-testid="specialEquipment"] [class*=RentzilaProposes_proposes_list] > * [class*=RentzilaProposes_name]';
+
+    async scrollToServicesSection(): Promise<void> {
+        const servicesSectionElement = $(this.servicesSection);
```
Comment (AndreyCar) [inline test/pageobjects/home.page.ts]: You can replace it with
```typescript
await $(this.servicesSection).scrollIntoView();
```

File: test/pageobjects/page.ts [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+
+    get errorPopupLoginMessages() {
+        return $$('[class*=CustomReactHookInput_error_message]');
+    }
+
+    get errorPopupLoginInputFields() {
+        return $$('[class*="CustomReactHookInput_error_input"]');
+    }
+
+    readonly footer = 'div[class*="Footer_footer"]';
```
Comment (AndreyCar) [inline test/pageobjects/page.ts]: I think you should use or selector or get.

File: test/specs/authorizationWithEmptyFields.e2e.ts
Comment (AndreyCar) [inline test/specs/authorizationWithEmptyFields.e2e.ts:1]: You should follow the next pattern when automating a test case:
one test case -> one file -> one it block
So, move all tests in this file to one it block

File: test/specs/authorizationWithInvalidEmail.e2e.ts
Comment (AndreyCar) [inline test/specs/authorizationWithInvalidEmail.e2e.ts:1]: Here is the same

File: test/specs/authorizationWithInvalidPassword.e2e.ts
Comment (AndreyCar) [inline test/specs/authorizationWithInvalidPassword.e2e.ts:1]: And here

File: test/specs/authorizationWithInvalidPhone.e2e.ts
Comment (AndreyCar) [inline test/specs/authorizationWithInvalidPhone.e2e.ts:1]: And here

File: test/specs/authorizationWithValidEmailAndPassword.e2e.ts
Comment (AndreyCar) [inline test/specs/authorizationWithValidEmailAndPassword.e2e.ts:1]: Also here

File: test/specs/authorizationWithValidPhoneAndPassword.e2e.ts
Comment (AndreyCar) [inline test/specs/authorizationWithValidPhoneAndPassword.e2e.ts:1]: The same

File: test/specs/checkServicesSection.e2e.ts
Comment (AndreyCar) [inline test/specs/checkServicesSection.e2e.ts:1]: And here

File: test/specs/verifyAllElementsOnTheFooterDisplayed.e2e.ts
Comment (AndreyCar) [inline test/specs/verifyAllElementsOnTheFooterDisplayed.e2e.ts:1]: Also here

File: test/specs/verifyDoYouHaveQuestionsForm.e2e.ts
Comment (AndreyCar) [inline test/specs/verifyDoYouHaveQuestionsForm.e2e.ts:1]: Also here

File: test/specs/сheckSpecialEquipmentSection.e2e.ts
Comment (AndreyCar) [inline test/specs/сheckSpecialEquipmentSection.e2e.ts:1]: Also here

File: wdio.conf.ts
Comment (AndreyCar) [inline wdio.conf.ts:1]: Create a config for the headed mode (when you can see a browser)

File: test/specs/authorizationWithInvalidEmail.e2e.ts [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+describe('id:C576 - Authorization with invalid email', () => {
+    it('1. Enter invalid email with spaces', async () => {
+        await homePage.clickLoginButton();
+
+        await homePage.emailField.setValue('testuser  rentzila@gmail.com');
+
+        await homePage.submitButton.click();
+            
+        await expect(homePage.popupWindow).toBeDisplayed();
+        await expect(await homePage.emailFieldErrorMessage.getText()).toEqual('Неправильний формат email або номера телефону')
```
Comment (AndreyCar) [inline test/specs/authorizationWithInvalidEmail.e2e.ts]: Replace it with
`await expect(homePage.emailFieldErrorMessage).toHaveText('Неправильний формат email або номера телефону')`

File: test/specs/authorizationWithEmptyFields.e2e.ts [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+require('dotenv').config();
+
+describe('id:C200 - Authorization with empty fields', () => {
+    it('1. Click on the "Увійти" at the bottom of the form with all fields empty.', async () => {
+        await homePage.clickLoginButton();
+        await homePage.submitButton.click();
+
+        await expect(browser).toHaveUrl('https://dev.rentzila.com.ua/');
+        await expect(homePage.popupWindow).toBeDisplayed();
+        await expect(await homePage.emailField.getAttribute('class')).toContain('CustomReactHookInput_error_input');
```
Comment (AndreyCar) [inline test/specs/authorizationWithEmptyFields.e2e.ts]: Please, dont use static expects (toBe, toContain, etc). WDIO has toHaveAttr() that waits for the attribute to equal the value

File: test/specs/authorizationWithEmptyFields.e2e.ts [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+        await expect(homePage.emailFieldErrorMessage).toBeDisplayed();
+    });
+
+    it('4. Enter any valid password into the ""Пароль"" Input, then click on the ""Увійти"" at the bottom of the form', async () => {
+        await homePage.passwordField.setValue(`${process.env.ADMIN_PASSWORD}`);
+
+        await homePage.submitButton.click();
+
+        await expect(browser).toHaveUrl('https://dev.rentzila.com.ua/');
```
Comment (AndreyCar) [inline test/specs/authorizationWithEmptyFields.e2e.ts]: Use files with constants

**Review by AndreyCar — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (borderForNoone) [reply package.json]: fixed

File: package.json (hunk 2 above)
Comment (borderForNoone) [reply package.json]: fixed

File: package.json (hunk 3 above)
Comment (borderForNoone) [reply package.json]: fixed

File: test/helpers/apiHelper.ts (hunk 4 above)
Comment (borderForNoone) [reply test/helpers/apiHelper.ts]: done

File: test/helpers/apiHelper.ts (hunk 5 above)
Comment (borderForNoone) [reply test/helpers/apiHelper.ts]: done

File: test/helpers/contactUsHelper.ts
Comment (borderForNoone) [reply test/helpers/contactUsHelper.ts:1]: done

File: test/pageobjects/adverts.page.ts [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+            const nameElement = unit.$(this.relevantUnitNames);
+            return nameElement.getText();
+        });
+
+        const names = await Promise.all(namesPromises);
+        return names;
+    }
+
+    async getAllCategoriesUnitsNames(): Promise<string[]> {
+        const products = $$(this.relevantCategoriesUnits);
```
Comment (AndreyCar) [inline test/pageobjects/adverts.page.ts]: Create a method for this in the page.ts

File: test/specs/authorizationWithEmptyFields.e2e.ts (hunk 13 above)
Comment (AndreyCar) [reply test/specs/authorizationWithEmptyFields.e2e.ts]: RegExp(`CustomReactHookInput_${test}_input`)
RegExp(erroText)

File: wdio.conf.ts
Comment (AndreyCar) [reply wdio.conf.ts:1]: import { config } from './wdio.conf';

config.capabilities[0]['goog:chromeOptions'].args = ['disable-gpu'];

exports.config = config;

**Review by AndreyCar — CHANGES_REQUESTED** [review]

File: test/helpers/contactUsHelper.ts (hunk 6 above)
Comment (borderForNoone) [reply test/helpers/contactUsHelper.ts]: done

File: test/helpers/contactUsHelper.ts (hunk 7 above)
Comment (borderForNoone) [reply test/helpers/contactUsHelper.ts]: done

File: test/helpers/contactUsHelper.ts (hunk 8 above)
Comment (borderForNoone) [reply test/helpers/contactUsHelper.ts]: done

File: test/pageobjects/adverts.page.ts (hunk 9 above)
Comment (borderForNoone) [reply test/pageobjects/adverts.page.ts]: done

File: test/pageobjects/home.page.ts (hunk 10 above)
Comment (borderForNoone) [reply test/pageobjects/home.page.ts]: done

File: test/specs/authorizationWithEmptyFields.e2e.ts
Comment (borderForNoone) [reply test/specs/authorizationWithEmptyFields.e2e.ts:1]: done

File: test/pageobjects/home.page.ts (hunk 10 above)
Comment (borderForNoone) [reply test/pageobjects/home.page.ts]: done

File: test/specs/authorizationWithInvalidEmail.e2e.ts
Comment (borderForNoone) [reply test/specs/authorizationWithInvalidEmail.e2e.ts:1]: done

File: test/specs/authorizationWithInvalidPhone.e2e.ts
Comment (borderForNoone) [reply test/specs/authorizationWithInvalidPhone.e2e.ts:1]: done

File: test/specs/authorizationWithInvalidPassword.e2e.ts
Comment (borderForNoone) [reply test/specs/authorizationWithInvalidPassword.e2e.ts:1]: done

File: test/specs/authorizationWithValidEmailAndPassword.e2e.ts
Comment (borderForNoone) [reply test/specs/authorizationWithValidEmailAndPassword.e2e.ts:1]: done

File: test/specs/authorizationWithValidPhoneAndPassword.e2e.ts
Comment (borderForNoone) [reply test/specs/authorizationWithValidPhoneAndPassword.e2e.ts:1]: done

File: test/specs/checkServicesSection.e2e.ts
Comment (borderForNoone) [reply test/specs/checkServicesSection.e2e.ts:1]: done

File: test/specs/verifyAllElementsOnTheFooterDisplayed.e2e.ts
Comment (borderForNoone) [reply test/specs/verifyAllElementsOnTheFooterDisplayed.e2e.ts:1]: done

File: test/specs/verifyDoYouHaveQuestionsForm.e2e.ts
Comment (borderForNoone) [reply test/specs/verifyDoYouHaveQuestionsForm.e2e.ts:1]: done

File: test/specs/сheckSpecialEquipmentSection.e2e.ts
Comment (borderForNoone) [reply test/specs/сheckSpecialEquipmentSection.e2e.ts:1]: done

File: wdio.conf.ts
Comment (borderForNoone) [reply wdio.conf.ts:1]: done

File: test/specs/authorizationWithInvalidEmail.e2e.ts (hunk 12 above)
Comment (borderForNoone) [reply test/specs/authorizationWithInvalidEmail.e2e.ts]: done

File: test/specs/authorizationWithEmptyFields.e2e.ts (hunk 13 above)
Comment (borderForNoone) [reply test/specs/authorizationWithEmptyFields.e2e.ts]: done

File: test/specs/authorizationWithEmptyFields.e2e.ts (hunk 14 above)
Comment (borderForNoone) [reply test/specs/authorizationWithEmptyFields.e2e.ts]: done

File: test/helpers/apiHelper.ts
Comment (AndreyCar) [inline test/helpers/apiHelper.ts:1]: Move it to test/api/rentzilla.api.ts

File: test/helpers/apiHelper.ts [hunk 16]
```diff
@@ -0,0 +1,69 @@
+import axios, { AxiosInstance } from 'axios';
+
+class ApiService {
+    private adminAccessToken: string | null = null;
+    private readonly axiosInstance: AxiosInstance;
+    private readonly apiUrl: string = `${process.env.BASE_URL}api`;
```
Comment (AndreyCar) [inline test/helpers/apiHelper.ts]: `${process.env.BASE_URL}api` -> `${process.env.BASE_URL}/api`

File: test/specs/autorization/authorizationWithEmptyFields.e2e.ts [hunk 17]
```diff
@@ -0,0 +1,47 @@
+import { expect } from '@wdio/globals'
+import homePage from '../../pageobjects/home.page';
+
+describe('id:C200 - Authorization with empty fields', () => {
+    it('1. Click on the "Увійти" at the bottom of the form with all fields empty.', async () => {
+        await homePage.clickLoginButton();
+        await homePage.submitButton.click();
+
+        await expect(browser).toHaveUrl(`${process.env.BASE_URL}`);
```
Comment (AndreyCar) [inline test/specs/autorization/authorizationWithEmptyFields.e2e.ts:11]: `'${process.env.BASE_URL}'` -> `process.env.BASE_URL`

File: test/specs/autorization/authorizationWithEmptyFields.e2e.ts [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+        await expect(homePage.popupWindow).toBeDisplayed();
+        await expect(homePage.emailField).toHaveAttr('class', /CustomReactHookInput_error_input/);
+        await expect(homePage.passwordField).toHaveAttr('class', /CustomReactHookInput_error_input/);
+        await expect(homePage.emailFieldErrorMessage).toBeDisplayed();
+        await expect(homePage.passwordFieldErrorMessage).toBeDisplayed();
+    });
+
+    it('2. Enter valid email into the ""E-mail або номер телефону"", then click on the ""Увійти"" at the bottom of the form', async () => {
+        await homePage.emailField.setValue(`${process.env.ADMIN_USERNAME}`);
```
Comment (AndreyCar) [inline test/specs/autorization/authorizationWithEmptyFields.e2e.ts:18]: `'${process.env.ADMIN_USERNAME}'` -> `process.env.ADMIN_USERNAME`

File: test/specs/autorization/authorizationWithEmptyFields.e2e.ts [hunk 19]
```diff
@@ -0,0 +1,47 @@
+import { expect } from '@wdio/globals'
+import homePage from '../../pageobjects/home.page';
+
+describe('id:C200 - Authorization with empty fields', () => {
+    it('1. Click on the "Увійти" at the bottom of the form with all fields empty.', async () => {
```
Comment (AndreyCar) [inline test/specs/autorization/authorizationWithEmptyFields.e2e.ts]: Merge it in to one it block

File: test/specs/autorization/authorizationWithInvalidEmail.e2e.ts [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+describe('id:C576 - Authorization with invalid email', () => {
+    it('1. Enter invalid email with spaces', async () => {
+        await homePage.clickLoginButton();
+
+        await homePage.emailField.setValue('testuser  rentzila@gmail.com');
+
+        await homePage.submitButton.click();
+            
+        await expect(homePage.popupWindow).toBeDisplayed();
+        await expect(homePage.emailFieldErrorMessage).toHaveText('Неправильний формат email або номера телефону')
```
Comment (AndreyCar) [inline test/specs/autorization/authorizationWithInvalidEmail.e2e.ts]: Move the text to costants

File: test/specs/footer/verifyAllElementsOnTheFooterDisplayed.e2e.ts [hunk 21]
```diff
@@ -0,0 +1,122 @@
+import { expect } from '@wdio/globals'
+import homePage from '../../pageobjects/home.page';
+import privacyPolicyPage from '../../pageobjects/privacyPolicy.page';
+import cookiePolicyPage from '../../pageobjects/cookiePolicy.page';
+import termsConditionsPage from '../../pageobjects/termsConditions.page';
+import advertsPage from '../../pageobjects/adverts.page';
+import tendersPage from '../../pageobjects/tenders.page';
+
+describe('id:C214 - Verify that all elements on the footer are displayed and all links are clickable', () => {
+    it('1. Scroll down to the footer', async () => {
```
Comment (AndreyCar) [inline test/specs/footer/verifyAllElementsOnTheFooterDisplayed.e2e.ts]: Use one it-block for the entire test. Do not split it with it blocks

**Review by AndreyCar — CHANGES_REQUESTED** [review]

File: test/pageobjects/page.ts (hunk 11 above)
Comment (borderForNoone) [reply test/pageobjects/page.ts]: done

File: test/pageobjects/adverts.page.ts (hunk 15 above)
Comment (borderForNoone) [reply test/pageobjects/adverts.page.ts]: done

File: test/helpers/apiHelper.ts
Comment (borderForNoone) [reply test/helpers/apiHelper.ts:1]: done

File: test/helpers/apiHelper.ts (hunk 16 above)
Comment (borderForNoone) [reply test/helpers/apiHelper.ts]: @AndreyCar, it doesn't work like that, the case of jurl ends like this: ua/

File: test/specs/autorization/authorizationWithEmptyFields.e2e.ts (hunk 17 above)
Comment (borderForNoone) [reply test/specs/autorization/authorizationWithEmptyFields.e2e.ts:11]: @AndreyCar, error:
image

File: test/specs/autorization/authorizationWithEmptyFields.e2e.ts (hunk 18 above)
Comment (borderForNoone) [reply test/specs/autorization/authorizationWithEmptyFields.e2e.ts:18]: https://github.com/borderForNoone/WebdriverIOTypeScriptRentzila/pull/3#discussion_r1786292390

File: test/specs/autorization/authorizationWithEmptyFields.e2e.ts (hunk 19 above)
Comment (borderForNoone) [reply test/specs/autorization/authorizationWithEmptyFields.e2e.ts]: done

File: test/specs/autorization/authorizationWithInvalidEmail.e2e.ts (hunk 20 above)
Comment (borderForNoone) [reply test/specs/autorization/authorizationWithInvalidEmail.e2e.ts]: done

File: test/specs/footer/verifyAllElementsOnTheFooterDisplayed.e2e.ts (hunk 21 above)
Comment (borderForNoone) [reply test/specs/footer/verifyAllElementsOnTheFooterDisplayed.e2e.ts]: done

File: mochawesome-reports/wdio-0-0-mochawesome-reporter.log
Comment (AndreyCar) [inline mochawesome-reports/wdio-0-0-mochawesome-reporter.log:1]: Add mochawesome-reports folder to .gitignore

File: test/helpers/serviceHelper.ts [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+    const serviceNames = await homePage.getAllItemNames();
+    
+    for (let i = 0; i < serviceNames.length; i++) {
+        await homePage.clickServiceTab(serviceLocator);
+        await homePage.scrollToServicesSection();
+        await homePage.validateServicesSection();
+
+        const currentServiceName = serviceNames[i];
+        const serviceItems = $$(homePage.serviceItems);
+        await serviceItems[i].click();
```
Comment (AndreyCar) [inline test/helpers/serviceHelper.ts]: create a method in the Homa page for this
```
async secviceItemClickByIndex(index: numbe){
  await $$(this,.serviceItems)[index].click()
}
```

File: test/helpers/serviceHelper.ts [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+export async function repeatTestCaseForspecialEquipment(serviceLocator: string) { 
+    await homePage.clickServiceTab(serviceLocator);
+
+    const serviceNames = await homePage.getAllSpecialEquipmentItemsNames();
+    
+    for (let i = 0; i < serviceNames.length; i++) {
+        await homePage.clickServiceTab(serviceLocator);
+
+        const currentServiceName = serviceNames[i];
+        const specialEquipmentItems = $$(homePage.specialEquipmentItems);
```
Comment (AndreyCar) [inline test/helpers/serviceHelper.ts]: the same

**Review by AndreyCar — APPROVED** [review]

**Review by AndreyCar — CHANGES_REQUESTED** [review]
.

File: mochawesome-reports/wdio-0-0-mochawesome-reporter.log
Comment (borderForNoone) [reply mochawesome-reports/wdio-0-0-mochawesome-reporter.log:1]: done

File: test/helpers/serviceHelper.ts (hunk 22 above)
Comment (borderForNoone) [reply test/helpers/serviceHelper.ts]: done

File: test/helpers/serviceHelper.ts (hunk 23 above)
Comment (borderForNoone) [reply test/helpers/serviceHelper.ts]: done

**Review by AndreyCar — APPROVED** [review]

**Pull request merged**